### ISSUE TITLE
Add Auto Moderation support.

### ISSUE THREAD

**Issue by paperdave** [CRBT-Team/Purplet#35]
I'm making this a separate issue because I feel like our rest client will be done before the issue I'm waiting on https://github.com/discordjs/discord-api-types/pull/418 to get merged.

**Comment by Clembs** [comment]
discord-api-types got this PR merged, we should update it across our packages and we should be good. closing as completed.

**Comment by paperdave** [comment]
its not completed until rest client ships with the routes that we have commented out and those types are confirmed.

**Comment by paperdave** [comment]
actually, i am wrong somewhat. we have types generated already, but i would really like to review before closing the issue. not sure if you did that already. if theres changes to be made to the generated types file let me know. (actually we should really re-run that generator in case other api changes have happened).
